clblas-cuda install failure due to nvidiatoolkit providing libOpenCL.so that refers to libOpenCL.so.1 but doesn't provide it

## Issue description

When I try and install the `clblas-cuda` package it fails because the installation procedure iterates over all the test-short executable's dependencies and is unable to locate the required *libOpenCL.so.1*.

It seems this dependency comes from the *test-short* binary being linked against the *libOpenCL.so* that comes with `nvidiatoolkit`.  This *libOpenCL.so*  has a `SONAME` record that says it is *libOpenCL.so.1*.

```bash
readelf -d /nix/store/vikagmf2qdni5mw0rk9lx9v441k7q60p-cudatoolkit-7.5.18/lib64/libOpenCL.so
```
```
...
 0x000000000000000e (SONAME)             Library soname: [libOpenCL.so.1]
...
```

This results in *test-short* requiring the library *libOpenCL.so.1*.  The `nvidiatoolkit` only provides a *libOpenCL.so* though, so the installation then explodes when it tries to loop over the dependencies of the binary it just created and finds it can't find *libOpenCL.so.1*.

It is not clear to me if this would be considered an issue with `clblas-cuda` or `nvidiatoolkit`.  Two possibly fixes are

1. add `ocl-icd` to `buildInputs` for `clblas-cuda` before `cudatoolkit`, or
2. add a `postFIxup` step to `nvidiatoolkit` to provide the *libOpenCL.so.1* symlink.

### Steps to reproduce

Try and install `clblas-cuda` package

```bash
nix-build '<nixpkgs>' -A clblas-cuda
```
```
...
-- Defined OPENCL_ROOT: /nix/store/vikagmf2qdni5mw0rk9lx9v441k7q60p-cudatoolkit-7.5.18, ENV{OPENCL_ROOT}: 
-- OPENCL_INCLUDE_DIRS: /nix/store/vikagmf2qdni5mw0rk9lx9v441k7q60p-cudatoolkit-7.5.18/include
-- Looking for CL_VERSION_2_0
-- Looking for CL_VERSION_2_0 - not found
-- Looking for CL_VERSION_1_2
-- Looking for CL_VERSION_1_2 - found
-- Looking for CL_VERSION_1_1
-- Looking for CL_VERSION_1_1 - found
-- FindOpenCL searching for 64-bit libraries
-- Found OPENCL: /nix/store/vikagmf2qdni5mw0rk9lx9v441k7q60p-cudatoolkit-7.5.18/lib64/libOpenCL.so (found version "1.2") 
-- FindOpenCL /nix/store/vikagmf2qdni5mw0rk9lx9v441k7q60p-cudatoolkit-7.5.18/lib64/libOpenCL.so, /nix/store/vikagmf2qdni5mw0rk9lx9v441k7q60p-cudatoolkit-7.5.18/include
...
-- Installing: /nix/store/s7s3n0p8b2m5ig94z5almfa8v3fr01ln-clblas-cuda-git-20160505/bin/test-short
-- 
warning: cannot resolve item 'libOpenCL.so.1'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file 'libOpenCL.so.1' returning type 'other' -- possibly incorrect
...
CMake Error at tests/copyTestDependencies.cmake:97 (file):
  file INSTALL cannot find
  "/tmp/nix-build-clblas-cuda-git-20160505.drv-0/clBLAS-d20977ec4389c6b3751e318779410007c5e272f8-src/Build/libOpenCL.so.1".
Call Stack (most recent call first):
  tests/cmake_install.cmake:66 (include)
  cmake_install.cmake:76 (include)


make: *** [Makefile:84: install] Error 1
builder for ‘/nix/store/m79qvzbd8vdv5m9j2fax9c7nlkhc96ls-clblas-cuda-git-20160505.drv’ failed with exit code 2
error: build of ‘/nix/store/m79qvzbd8vdv5m9j2fax9c7nlkhc96ls-clblas-cuda-git-20160505.drv’ failed
```

## Technical details

* System:

```
LSB Version:	:base-4.0-amd64:base-4.0-noarch:core-4.0-amd64:core-4.0-noarch:graphics-4.0-amd64:graphics-4.0-noarch:printing-4.0-amd64:printing-4.0-noarch
Distributor ID:	CentOS
Description:	CentOS release 6.8 (Final)
Release:	6.8
Codename:	Final
```

* Nix version: `nix-env (Nix) 1.11.4`
* Nixpkgs version: `17.03.325.73de4ca`
* Sandboxing enabled: `build-use-sandbox = true`

I decided to give the latest unstable a try as well.  It dies during the `configurePhase` as it cannot find *libOpenCL.so* at all anymore.

```bash
nix-build '<nixpkgs>' -A clblas-cuda
```
```
...
-- Defined OPENCL_ROOT: /nix/store/lb8yzw0xsw7kd2ahzjsa7dgyhgll5n62-cudatoolkit-7.5.18, ENV{OPENCL_ROOT}: 
-- OPENCL_INCLUDE_DIRS: /nix/store/lb8yzw0xsw7kd2ahzjsa7dgyhgll5n62-cudatoolkit-7.5.18/include
-- Looking for CL_VERSION_2_0
-- Looking for CL_VERSION_2_0 - not found
-- Looking for CL_VERSION_1_2
-- Looking for CL_VERSION_1_2 - found
-- Looking for CL_VERSION_1_1
-- Looking for CL_VERSION_1_1 - found
-- FindOpenCL searching for 64-bit libraries
-- Could NOT find OPENCL (missing:  OPENCL_LIBRARIES) (found version "1.2")
-- FindOpenCL looked for libraries named: OpenCL
...
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
OPENCL_LIBRARIES (ADVANCED)
...
-- Configuring incomplete, errors occurred!
See also "/tmp/nix-build-clblas-cuda-git-20160505.drv-0/clBLAS-d20977ec4389c6b3751e318779410007c5e272f8-src/Build/CMakeFiles/CMakeOutput.log".
See also "/tmp/nix-build-clblas-cuda-git-20160505.drv-0/clBLAS-d20977ec4389c6b3751e318779410007c5e272f8-src/Build/CMakeFiles/CMakeError.log".
builder for ‘/nix/store/ki7pirlcxs9317s5ras52cw32izikdr9-clblas-cuda-git-20160505.drv’ failed with exit code 1
error: build of ‘/nix/store/ki7pirlcxs9317s5ras52cw32izikdr9-clblas-cuda-git-20160505.drv’ failed
```

Looking through the [nvidiatoolkit](https://github.com/NixOS/nixpkgs/blob/32033f5bf300f0d94a786983110270fb8e053246/pkgs/development/compilers/cudatoolkit/default.nix#L83) source it looks like the recent commit 044a49f stripped out *libOpenCL.so* all together.  I guess that makes (1) above the correct solution.

--- a/pkgs/development/libraries/science/math/clblas/cuda/default.nix
+++ b/pkgs/development/libraries/science/math/clblas/cuda/default.nix
@@ -5,6 +5,7 @@
 , blas
 , boost
 , python
+, ocl-icd
 , cudatoolkit
 , nvidia_x11
 , gtest
@@ -51,6 +52,7 @@ stdenv.mkDerivation rec {
     gfortran
     blas
     python
+    ocl-icd
     cudatoolkit
     nvidia_x11
     gtest

